### PR TITLE
Implement -webkit-radial-gradient() (fixes #15441)

### LINKED ISSUE #15441: Implement parsing webkit-prefixed gradient functions
This is defined in the Compatibility Standard: https://compat.spec.whatwg.org/#css-image-type

The corresponding syntax is in https://www.w3.org/TR/2011/WD-css3-images-20110217/#gradients

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16511)

### CODE REVIEW

**Comment by jdm** [comment]
```
---- parsing::image::test_radial_gradient stdout ----
	thread 'parsing::image::test_radial_gradient' panicked at 'Failed to parse radial-gradient(circle closest-side at 20px 30px, red, green): ()', src\libcore\result.rs:859
```

**Comment by nox** [comment]
Forgot some cases in my first commit.

File: components/style/values/specified/image.rs [hunk 1]
```diff
@@ -358,6 +326,40 @@ fn parse_position(context: &ParserContext, input: &mut Parser) -> Result<Positio
...
+    } else if let Ok(keyword) = input.try(SizeKeyword::parse) {
+        // Handle <keyword> <shape-keyword>?
+        if input.try(|input| input.expect_ident_matching("circle")).is_ok() {
+            Ok(EndingShape::Circle(LengthOrKeyword::Keyword(keyword)))
+        } else {
+            let _ = input.try(|input| input.expect_ident_matching("ellipse"));
+            Ok(EndingShape::Ellipse(LengthOrPercentageOrKeyword::Keyword(keyword)))
+        }
+    } else {
+        if input.try(|input| input.expect_ident_matching("ellipse")).is_ok() {
```
Comment (emilio) [inline components/style/values/specified/image.rs]: Any particular reason why this isn't merged with the above `} else {`?

File: components/style/gecko/conversions.rs [hunk 2]
```diff
@@ -216,6 +216,8 @@ impl nsStyleImage {
                                     SizeKeyword::FarthestSide => NS_STYLE_GRADIENT_SIZE_FARTHEST_SIDE,
                                     SizeKeyword::ClosestCorner => NS_STYLE_GRADIENT_SIZE_CLOSEST_CORNER,
                                     SizeKeyword::FarthestCorner => NS_STYLE_GRADIENT_SIZE_FARTHEST_CORNER,
+                                    SizeKeyword::Contain => NS_STYLE_GRADIENT_SIZE_CLOSEST_SIDE,
+                                    SizeKeyword::Cover => NS_STYLE_GRADIENT_SIZE_FARTHEST_CORNER,
```
Comment (emilio) [inline components/style/gecko/conversions.rs]: Can we move these to a helper and reuse them? Feel free to file an e-easy issue if you prefer.

File: components/style/values/specified/image.rs [hunk 3]
```diff
@@ -254,6 +276,25 @@ impl GradientKind {
 
         Ok(GradientKind::Radial(shape, position))
     }
+
+    /// Parses a webkit radial gradient from the given arguments.
```
Comment (emilio) [inline components/style/values/specified/image.rs]: A link to the compat spec would be nice :)

**Review by emilio — COMMENTED** [review]
r=me w/ nits, thanks!

**Pull request merged**